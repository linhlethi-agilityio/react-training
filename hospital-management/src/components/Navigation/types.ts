interface NavigationConfig {
  id: number;
  name: string;
  path: string;
}

export interface INavigationProps {
  items: NavigationConfig[];
}
