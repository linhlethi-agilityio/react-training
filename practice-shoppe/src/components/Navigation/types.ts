interface NavigationConfig {
  id: number;
  name: string;
  path: string;
}

export interface INavProps {
  className?: string;
  items: NavigationConfig[];
}
