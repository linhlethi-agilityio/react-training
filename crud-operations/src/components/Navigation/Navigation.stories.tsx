import type { Meta, StoryObj } from '@storybook/react';

// Components
import Navigation from '.';

const meta: Meta<typeof Navigation> = {
  component: Navigation,
};

export default meta;
type Story = StoryObj<typeof Navigation>;

export const Default: Story = {
  args: {
    isShorter: false,
  },
};

export const Shorter: Story = {
  args: {
    isShorter: true,
  },
};
