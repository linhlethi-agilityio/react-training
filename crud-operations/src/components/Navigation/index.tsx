import { memo } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
import { VStack, Icon } from '@chakra-ui/react';

// Constants
import { SIDEBAR_NAVIGATION } from '@constants';

// Components
import { SideBarItem } from '@components';

interface NavigationProps {
  isShorter?: boolean;
}

const Navigation = ({ isShorter = false }: NavigationProps) => {
  const navigate = useNavigate();
  const location = useLocation();

  return (
    <VStack
      spacing={isShorter ? 4 : 18}
      justifyContent="flex-start"
      alignItems="flex-start"
      w={isShorter ? '100%' : 193}
      transition="1s"
    >
      {SIDEBAR_NAVIGATION.map(({ label, icon, router }) => {
        const isFocused = router === location.pathname;

        return (
          <SideBarItem
            key={label}
            isFocused={isFocused}
            icon={<Icon as={icon} />}
            label={!isShorter && label}
            onClick={() => navigate(router)}
          />
        );
      })}
    </VStack>
  );
};

export default memo(Navigation);
