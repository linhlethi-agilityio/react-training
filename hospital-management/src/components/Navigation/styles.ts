import { createStyles } from '@mantine/core';

const useStyles = createStyles((theme) => ({
  nav: {
    display: 'flex',
  },

  navItem: {
    color: theme.colors.dark[3],
    textDecoration: 'none',
  },
}));

export { useStyles };
