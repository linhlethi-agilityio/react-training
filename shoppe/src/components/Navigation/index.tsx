import { memo, ReactElement } from 'react';
import styled from 'styled-components';
import { INavProps } from './types';
import { StyledNavigationLink } from '@components/Link';

const Navigation = (props: INavProps): ReactElement => {
  const { className, items } = props;

  return (
    <nav className={className}>
      {items.map((item) => (
        <StyledNavigationLink key={item.id} to={item.path}>
          {item.name}
        </StyledNavigationLink>
      ))}
    </nav>
  );
};

const StyledNavigation = memo(styled(Navigation)<INavProps>`
  gap: 41px;
  display: flex;
  margin-top: 15px;
`);

export { StyledNavigation };
