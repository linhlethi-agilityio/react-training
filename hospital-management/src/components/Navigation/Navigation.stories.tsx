import { ComponentMeta, ComponentStory } from '@storybook/react';
import { StyledNavigation } from '@components/Navigation';
import { NAVIGATION_ITEM } from '@constants/baseConfig';
import { BrowserRouter as Router } from 'react-router-dom';

export default {
  title: 'Navigation',
  component: StyledNavigation,
} as ComponentMeta<typeof StyledNavigation>;

const Template: ComponentStory<typeof StyledNavigation> = (args) => (
  <Router>
    <StyledNavigation {...args} />
  </Router>
);

export const Navigation = Template.bind({});
const navigation = {
  items: NAVIGATION_ITEM,
};
Navigation.args = navigation;
