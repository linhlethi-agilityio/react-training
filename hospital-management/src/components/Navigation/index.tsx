import { Link } from 'react-router-dom';
import { INavigationProps } from './types';
import { memo } from 'react';
import { useStyles } from './styles';
import { NavLink } from '@mantine/core';

const Navigation = (props: INavigationProps) => {
  const { items } = props;
  const { classes } = useStyles();

  return (
    <nav className={classes.nav}>
      {items.map((item) => (
        <NavLink
          className={classes.navItem}
          component={Link}
          to={item.path}
          key={item.id}
          label={item.name}
        />
      ))}
    </nav>
  );
};

export const MemoizedNavigation = memo(Navigation);
