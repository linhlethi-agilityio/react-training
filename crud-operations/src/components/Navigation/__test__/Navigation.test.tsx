import { render } from '@test-utils';
import '@testing-library/jest-dom';
import { MemoryRouter, useLocation } from 'react-router-dom';

// Components
import Navigation from '..';

const mockProps = {
  isShorter: false,
};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
  useLocation: () => jest.fn(),
}));

// Mock the constants module
jest.mock('@constants', () => ({
  SIDEBAR_NAVIGATION: [
    {
      label: 'Home',
      router: '/',
    },
    {
      label: 'Course',
      router: '/course',
    },
  ],
}));

describe('Navigation component', () => {
  test('should match snapshot for Navigation', () => {
    const { container } = render(<Navigation {...mockProps} />);

    expect(container).toMatchSnapshot();
  });

  it.skip('renders navigation items correctly', () => {
    (useLocation as jest.Mock).mockReturnValue({ location: { pathname: '/' } });

    const { getByText } = render(
      <MemoryRouter>
        <Navigation />
      </MemoryRouter>,
    );

    expect(getByText('Home')).toBeInTheDocument();
    expect(getByText('Course')).toBeInTheDocument();
  });
});
