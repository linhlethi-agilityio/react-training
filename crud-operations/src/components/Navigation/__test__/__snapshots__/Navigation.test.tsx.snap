// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should match snapshot for Navigation 1`] = `
<div>
  <div
    class="chakra-stack css-1kae02p"
  />
</div>
`;
